### ISSUE TITLE
Search Statistics has more not calculated as it should be

### ISSUE THREAD

**Issue by Shaihuludus** [Adobe-Marketing-Cloud/asset-share-commons#259]
I have 33 assets in my search folder set guessTotal  set to 500 and limit to 24. With Search Statistics I would expected to have 24 founded, 33 total (without+)  and time as AEM thinks is ok. Unfortunately my results contains total: 33+

**Environment**
* AEM Version: 6.4.1
* Asset Share Commons Version: 1.6.3-SNAPSHOT

**To Reproduce**
1. Go to Search page and configure Search Results as mentioned above
2. Go to publish
3. 
4. In search statistics I have total 33+

**Expected behavior**
Should be total: 33

I don't know why in 
```
QueryBuilderResultsImpl.class
is
this.more = searchResult.hasMore() || this.runningTotal < searchResult.getTotalMatches();
```

This was checked with https://github.com/Adobe-Marketing-Cloud/asset-share-commons/pull/257

**Comment by Shaihuludus** [comment]
Hi, 
Is there any new information about this issue. I'm happy to fixed but I wouldn't like to break something. That's why I would need to know about this code fragment mentioned  above

**Comment by davidjgonzalez** [comment]
Hi @Shaihuludus - sorry for the tardy response.
The logic above is a weird case... If memory serves this is logic behind it:

* `.hasMore()` is really used with `guessTotal` (which we require for performance reasons). If guessTotal = 10, and w/ a total result set of 100, with a limit of 5, hasMore() will be true; since guessTotal will only count up to 10 results before quitting.
* `this.runningTotal < searchResult.getTotalMatches()` handles the case when guessTotal is > than the result set. So for instance, if you have a guessTotal = 1000, and the total result set for a query is 100, but your loading in sets of 25... then hasMore() will always be false, however after the first 25 results are loaded, there are still 75 "more" results to load.

https://helpx.adobe.com/experience-manager/6-4/sites/developing/using/reference-materials/javadoc/com/day/cq/search/result/SearchResult.html#hasMore()

**Comment by Shaihuludus** [comment]
Thanks, now I understand it's purpose. But it still causes the bug in case I described. Mayby solution would be to extend SearchResults to keep also original hasMore (the one from AEM). Such parameter could be used in SearchStatistics (and allow to display + correctly). The value calculated by you could be used as is now to calculate "load more"

I can prepare pr tommorow if You are okay with this solution.  (Time difference)

**Comment by Shaihuludus** [comment]
Thanks, now I understand. But it still causes bug in search statistics. Mayby solution would be to extend search result and pass also original hasMore from AEM. Then this value could be used  with statistics and Your original with load more.

If this solution is ok I can prepare PR tommorow (time difference)

**Comment by davidjgonzalez** [comment]
Seems reasonable to me assuming that is a reliable value to fix the problem (i took a look at this when you first log, but if im being honest, forget what i saw :))

PR away!

**Comment by davidjgonzalez** [comment]
released in v1.6.10
